### PR TITLE
Document how to run kubernetes-api demo on open shift

### PR DESCRIPTION
**Pull request by chbatey**
- Change to use a namespace for both minishift and minikube to make the
same resources work

Start of #365

### CODE REVIEW

**Comment by chbatey** [comment]
I've also tried to make the k8s demos more consistent under this PR

**Comment by chbatey** [comment]
And verified that the DNS one works on openshift

**Comment by chbatey** [comment]
Rebased, could you review @raboof ?

File: bootstrap-demo/kubernetes-dns/src/main/resources/application.conf [hunk 1]
```diff
@@ -10,10 +10,11 @@ akka {
   management {
     cluster.bootstrap {
       contact-point-discovery {
-        port-name = "management" # these are the defaults in bootstrap 0.18
+        port-name = "management"
         protocol = "tcp"
         service-name = "bootstrap-demo-kubernetes-dns-internal"
-        service-namespace = "default.svc.cluster.local"
+        # Can be removed once https://github.com/akka/akka/issues/25825 is implemented
```
Comment (raboof) [inline bootstrap-demo/kubernetes-dns/src/main/resources/application.conf]: :+1:

File: bootstrap-demo/kubernetes-dns/test.sh [hunk 2]
```diff
@@ -20,23 +21,20 @@ then
...
 if [ $i -eq 10 ]
 then
   echo "No 3 MemberUp log events found"
-  for POD in $(kubectl get pods | grep demo-kubernetes-dns | grep Running | awk '{ print $1 }')
-  do
-    echo "Logging for $POD"
-    kubectl logs $POD bootstrap-demo-kubernetes-dns
-  done
+  echo "=============================="
+  kubectl logs $POD
```
Comment (raboof) [inline bootstrap-demo/kubernetes-dns/test.sh]: I think in this case it is actually useful to collect the logging from all 3 pods, not just the first one, to diagnose what went wrong

Comment (raboof) [reply bootstrap-demo/kubernetes-dns/test.sh]: shame on me for treating `POD` as a `var` though, I guess :D

File: discovery-kubernetes-api/src/main/resources/reference.conf [hunk 3]
```diff
@@ -25,6 +25,7 @@ akka.discovery {
     #   valueFrom:
     #    fieldRef:
     #     fieldPath: metadata.namespace
+    # Can be removed once https://github.com/akka/akka-management/issues/345 is implemented
```
Comment (raboof) [inline discovery-kubernetes-api/src/main/resources/reference.conf]: :+1:

File: docs/src/main/paradox/bootstrap/kubernetes-api.md [hunk 4]
```diff
@@ -1,7 +1,5 @@
 # Kubernetes API
 
-An example project that can be deployed to Kubernetes via minikube is in `bootstrap-demo/kubernetes-api`.
```
Comment (raboof) [inline docs/src/main/paradox/bootstrap/kubernetes-api.md]: perhaps link back from this file and `kubernetes.md` to the kubernetes section of `recipes.md`?

Comment (chbatey) [reply docs/src/main/paradox/bootstrap/kubernetes-api.md]: :+1:

**Pull request merged**